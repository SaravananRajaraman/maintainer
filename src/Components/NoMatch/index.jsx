import React from 'react';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';

const StyledBox = styled('div')({
    display : 'grid',
    alignItems : 'center',
    justifyItems : 'center',
    height : '100vh'
})

export default function NoMatch() {
  return (
    <StyledBox data-test='NoMatch-Component'>
        <div>
            <Typography variant='h1' data-test='Error'>{'404'}</Typography>
            <Typography variant='h6' data-test='Description'>{'Page not found!'}</Typography>
        </div>        
    </StyledBox>
  );
};