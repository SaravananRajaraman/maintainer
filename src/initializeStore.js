import { createStore,applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import createReducer from "./reducers";

let middleware = [ thunk ];

const initializeStore = () => {
  
  const store = createStore(    
    createReducer(),       
    applyMiddleware(...middleware)     
  );   
  return store;
};

export default initializeStore;