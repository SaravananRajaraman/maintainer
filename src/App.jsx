import React from 'react';
import Header from './Components/Header';
import HomePage from './Components/HomePage';
import NoMatch from './Components/NoMatch';
import RepoTable from './Components/RepoTable';
import { connect } from "react-redux";
import {changeLoading,getRepos} from './actions';
import {HashRouter as Router, Route, Switch } from "react-router-dom";


class App extends React.Component{

  async componentDidMount(){
      await this.props.getRepos();      
      this.props.changeLoading();
  }

  render() {
    return (
      <React.Fragment>
          <Header/>
          <Router>
            <Switch>
              <Route exact path="/" component={HomePage}/>    
              <Route exact path="/Table" component={RepoTable}/>                              
              <Route path="*" component={NoMatch}/>                
            </Switch>
          </Router>          
      </React.Fragment>
    );
  };
};

export default connect(null,{changeLoading,getRepos})(App);