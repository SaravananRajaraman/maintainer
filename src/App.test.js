import React from 'react';
import { mount,shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import toJson from 'enzyme-to-json';
// import HomePage from './Components/HomePage';
import NoMatch from './Components/NoMatch';
// import RepoTable from './Components/RepoTable';
import initializeStore from './initializeStore';
import {findTestAttr} from './Utils';

const store = initializeStore();

import App from './App';

const setUp = (props={}) => {         
    const component = shallow(<App store={store} {...props} />);    
    return component.dive();
}

describe('App Component', () =>{

    it('Should render the component with no error', ()=>{   
        let   component = setUp();      
        let wrapper = findTestAttr(component,'App-Component');
        expect(wrapper.length).toBe(1);
    });

    it('invalid path should match screenShot',()=>{
        const wrapper = shallow(
            <MemoryRouter  initialEntries={[ '#/random' ]}>
              <App store={store}/>
            </MemoryRouter >
          ).dive().dive();                    
          expect(toJson(wrapper)).toMatchSnapshot();
    });

});
