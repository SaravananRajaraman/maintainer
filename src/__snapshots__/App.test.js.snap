// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component invalid path should match screenShot 1`] = `
<Connect(App)
  store={
    Object {
      "asyncReducers": Object {},
      "dispatch": [Function],
      "getState": [Function],
      "injectReducer": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
